Local / Remote plugin require is not consistent

### Expected behavior
Consistency while using `require` to use a Talk dependency in the plugin

### Actual behavior
`require` works differently depending if the plugin is local or remote.

### Steps to reproduce behavior
If you use the following plugin (https://github.com/mantika/sherloq-talk/blob/master/index.js) locally by copying the index.js in the proper folder, `require` works as expected. However, if you setup the plugin to be configured remotely through NPM (https://www.npmjs.com/package/sherloq-talk),  you get an error as `require` can't resolve imports.

```
talk_1   | Error: plugin 'sherloq-talk' could not be required from '/usr/src/app/node_modules/sherloq-ta
lk/index.js': Cannot find module 'services/comments.js'                                                 
talk_1   |     at PluginSection.plugins.itteratePlugins.map (/usr/src/app/plugins.js:104:23)            
talk_1   |     at Array.map (native)                                                                    
talk_1   |     at new PluginSection (/usr/src/app/plugins.js:91:45)                                     
talk_1   |     at new PluginManager (/usr/src/app/plugins.js:142:32)                                    
talk_1   |     at Object.<anonymous> (/usr/src/app/services/plugins.js:3:18)                            
talk_1   |     at Module._compile (module.js:571:32)                                                    
talk_1   |     at Object.Module._extensions..js (module.js:580:10)                                      
talk_1   |     at Module.load (module.js:488:32)                                                        
talk_1   |     at tryModuleLoad (module.js:447:12)                                                      
talk_1   |     at Function.Module._load (module.js:439:3)                                               
talk_1   |     at Module.require (module.js:498:17)                                                     
talk_1   |     at require (internal/module.js:20:19)                                                    
talk_1   |     at Object.<anonymous> (/usr/src/app/app.js:7:17)                                         
talk_1   |     at Module._compile (module.js:571:32)                                                    
talk_1   |     at Object.Module._extensions..js (module.js:580:10)                                      
talk_1   |     at Module.load (module.js:488:32)                                                        
talk_1   | /usr/src/app/plugins.js:105                                                                  
talk_1   |         throw e;                                                                             
talk_1   |         ^                                                                                    
talk_1   |                                                                                              

```

--- a/plugins.js
+++ b/plugins.js
@@ -3,9 +3,10 @@ const path = require('path');
 const resolve = require('resolve');
 const debug = require('debug')('talk:plugins');
 const Joi = require('joi');
+const amp = require('app-module-path');
 
-// Add support for require rewriting.
-require('app-module-path').addPath(__dirname);
+// Add the current path to the module root.
+amp.addPath(__dirname);
 
 let plugins = {};
 
@@ -35,6 +36,9 @@ try {
   }
 }
 
+/**
+ * All the hooks from plugins must match the schema defined here.
+ */
 const hookSchemas = {
   passport: Joi.func().arity(1),
   router: Joi.func().arity(1),
@@ -82,6 +86,12 @@ function pluginPath(name) {
   }
 }
 
+/**
+ * Itterates over the plugins and gets the plugin path's, version, and name.
+ *
+ * @param {Array<Object|String>} plugins
+ * @returns {Array<Object>}
+ */
 function itteratePlugins(plugins) {
   return plugins.map((p) => {
     let plugin = {};
@@ -110,6 +120,12 @@ function itteratePlugins(plugins) {
     return plugin;
   });
 }
+
+// Add each plugin folder to the allowed import path so that they can import our
+// internal dependancies.
+Object.keys(plugins).forEach((type) => itteratePlugins(plugins[type]).forEach((plugin) => {
+  amp.enableForDir(path.dirname(plugin.path));
+}));
 
 /**
  * Stores a reference to a section for a section of Plugins.

